### DISCUSSION TITLE
super Incremental Static Regeneration

### DISCUSSION THREAD

**Discussion by ats1999** [vercel/next.js#20195] [Help] [answered]
> Don't go with the title of this post
# Feature request
When we use  **Incremental Static Regeneration**, then we set the `revalidate` property to some number`(seconds)`. 
In my case, i am using 
```js
export async function getStaticProps() {
  const res = await getData(..get_data_from_database);

  return {
    props: {
      posts:JSON.stringify(res),
    },
    // - At most once every minute
    revalidate: 60, // In seconds
  }
}
```
So, here we are fetching data from the server every minute, whether data has been updated or not. NextJs will generate the new page only when data has been updated. Suppose, the size of a single document data is 0.5MB. So, when nextjs server will communicate with the database then it will fetch 0.5MB of data each time. 

Suppose my website has a lot of traffic, say 1000000000 requests in one minute. My website has 1000000 posts, and nextjs will fetch data about each post every minute, because of the `revalidate` option. 

Say, my posts are updated once in a week. It is also possible that a post can get updated 100 times in a single day. 

## `All of the above expiations are situation description, now i am going to explain actual problem`
For 1000000 post will communicate with the database 1000000 times in a single minute, 60000000 in an hour, 1440000000 in a day, and 43200000000. 

suppose, 100 post gets updated in a single minute, so 600 updated in an hour, 14400 updated in on a day and 432000 updated in a month.  

So, according to my calculation, we are wasting `43200000000-432000=43199568000` database query each month. This number could be much larger than this, but i kept it for example demonstration. We are also wasting `21599784000 MB` or `21093.5391 TB`  bandwidth each month. 

## Let's again take another simple example
The website has 1000000 posts, and 1000 posts can get updated in a single day. 
For 1000000 posts, the **next** server will communicate with the database 1440000000 times in a single day. 

## Solution
If we update the page only when data changes on the server. Or, we can trigger an update for a particular page. How can we do, this is a question. 
If anyhow we can do so, then we can save 1439999000 database queries and `703.124512 TB` bandwidth per day. 
> **Note:** The number could be much larger

**Comment by stigkj** [comment]
The developers have been talking about implementing this feature, i.e. let something external trigger a revalidate. It is not available in any form now as far as I know. Cannot see it in the experimental features either.

**Reply by ats1999** [reply]
It would be a great feature.

**Reply by times29** [reply]
We could use that as well!

**Reply by it-harrison** [reply]
I would love to have this feature

**Reply by abhikumar22** [reply]
Need this feature

**Reply by ats1999** [reply]
@abhikumar22 https://github.com/vercel/next.js/discussions/11552#discussioncomment-2655

**Reply by nradulaski-godaddy** [reply]
Me2.

**Comment by michrome** [comment]
> [We'll be exploring on-demand (triggered via API route) revalidation instead of interval-based in the future](https://github.com/vercel/next.js/discussions/11552#discussioncomment-2655).

**Reply by ats1999** [reply]
This would be really a great feature..

**Comment by vikasg603** [comment]
Any update on this issue?

**Reply by abhikumar22** [reply]
Eagerly waiting for this feature

**Comment by leerob** [comment, marked as answer]
I'm incredibly excited to share that Next.js 12.1 was just released, which includes **on-demand Incremental Static Regeneration**! We know many of you have been looking forward to this for a while, including myself, and I'm so happy it is here. 

- [Learn about Next.js 12.1](https://nextjs.org/blog/next-12-1)
- [Watch a demo of on-demand ISR](https://youtu.be/BGexHR1tuOA)
- [View another demo and give us your feedback](https://on-demand-isr.vercel.app/)
- [Read the documentation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)
